### DISCUSSION TITLE
Pulsar from Flatpak and packages with CLI usage

### DISCUSSION THREAD

**Discussion by AlexWayfer** [pulsar-edit/.github#271] [Core Application & Packages]
Hello.

I currently don't like Flatpak too much. It has weirder syntax to call (I have to create aliases (but they don't work with `sudo`)). It seems like a sandbox too much, Snap seems better system-integrated.

Firstly I've faced the issue with main menu color in GNOME. System settings don't affect Flatpak apps. Work-around: install Flatseal, set `GTK_THEME=Adwaita-dark` for Pulsar (and all other apps you want to), and, of course, install Adwaita Dark flatpak package (weird).

OK, it's resolvable. But now my Pulsar packages, which use CLI, just don't work. For example, I have `linter-rubocop` package, it tries to call RuboCop but can't see it. And I don't know what to do with it, how to pass my user env and shell config to Pulsar from Flatpak. I can set the absolute path, `/home/aleksandr/.rbenv/shims/rubocop`, but now I get `Linter-Rubocop: ruby: error while loading shared libraries: libcrypt.so.1: cannot open shared object file: No such file or directory`. So, I have to install additional system packages somehow into Flatpak Pulsar image, but I don't know how, and it seems like a double work (if not more, if I have to do it with multiple Flatpak apps).

`deb-get`, another way you distribute Pulsar, still doesn't work with Ubuntu 24.04 LTS.

So… how to pass my user's env to Flatpak app? Or do you have plans to distribute Pulsar in other ways? Like Snap, or PPA, or something else (with automatic updates).

**Comment by confused-Techie** [comment]
Thanks a ton for posting!

The general consensus among the team as far as I'm aware, is that there are some issues with the Flatpak, that's why it's not officially supported. The Pulsar team doesn't distribute the Flatpak version of Pulsar and that is entirely a community effort. If you'd like you can always feel free to open an issue with them to troubleshoot.

As for `deb-get` that one is officially supported, and apologies if I'm just out of the loop, but I wasn't aware that it didn't work on recent Ubuntu versions, I'd hope there's an issue tracker for that, but if not one should absolutely get created. 

Lastly, while we have no specific plans or goals for further distribution, generally we would like to get more officially supported methods of distribution. As far as I can see the major determiner if we can support it is the time and cost associated with doing so. 

Take for example `chocolatey` one of the community members wanted it to exist, and help set up an automated system to where we hardly have to spend any time at all for it, and now we are officially able to support it. So something similar is likely the best way forward if a community member wanted to assist in an area they are familiar with.

Otherwise if you'd like to contribute more to a specific avenue of distribution please check out this much more [in-depth discussion](https://github.com/orgs/pulsar-edit/discussions/77) on the topic.

**Reply by AlexWayfer** [reply]
> The Pulsar team doesn't distribute the Flatpak version of Pulsar and that is entirely a community effort. If you'd like you can always feel free to open an issue with them to troubleshoot.

Thank you. I didn't know that, I thought I saw it in official instructions, but I've messed Pulsar with EasyEffects, my bad. Anyway, my recommendation is not to use Flatpak for now officially, it has described problems with system-integration (and, as outcome, with packages), but of course it's up to you.

> As for deb-get that one is officially supported, and apologies if I'm just out of the loop, but I wasn't aware that it didn't work on recent Ubuntu versions, I'd hope there's an issue tracker for that, but if not one should absolutely get created.

Yeah, of course it has been already created, and even more — work has been done in `master`, as I understood. But we're waiting for a new release. https://github.com/wimpysworld/deb-get/issues/1081

> Otherwise if you'd like to contribute more to a specific avenue of distribution please check out this much more [in-depth discussion](https://github.com/orgs/pulsar-edit/discussions/77) on the topic.

I'm not an expert in distributing platforms, so I can't propose something to you (more than my personal experience), but I'll watch for progress and options there, thank you.

**Reply by confused-Techie** [reply]
Thanks a ton for the clarification that `deb-get` is having issues, I thought somehow Pulsar had broken when distributed via `deb-get` recently.

But absolutely keep an eye on there, and hopefully over time we will see more items to come. In the meantime hopefully `pulsar-updater` can serve as close to possible as an automatic update.
